### DISCUSSION TITLE
Add a Button/Action to Form in Admin Resource?

### DISCUSSION THREAD

**Discussion by yaplej** [filamentphp/filament#2632] [Help]
Hello,

Is it possible to add a button/action to my ModelResource Form so the button can be pressed and perform the defined action?  That would allow the button to be hidden/disabled just like other parts of the form?

I found how to add buttons to the bottom of the Form by changing the EditModel page.  Those are just always enabled and not quite what I was looking for.

https://github.com/laravel-filament/filament/discussions/2434#discussioncomment-2823359 
```
    protected function getFormActions(): array
    {
        return [
            $this->getSaveFormAction(),
            Action::make('my button')
                ->action(fn () => $this->record->delete())
                ->requiresConfirmation(),
            $this->getCancelFormAction(),
        ];
    }
```

**Comment by danharrin** [comment]
You can pass `->hidden()` or `->disabled()` to the button if you want to control that?

**Reply by yaplej** [reply]
What about putting them in the form as fields so the placement can be manipulated?

**Reply by danharrin** [reply]
You can do that with View (layout) components in the schema

**Reply by yaplej** [reply]
I have looked through the documentation for both Actions and Layout but am still missing how to put the Action Buttons somewhere specific on the form.  Re-checked the Demo too but none of the examples had a custom action on the Form.
https://filamentphp.com/docs/2.x/forms/layout#getting-started

**Reply by danharrin** [reply]
Yes, there is no documentation as it is a custom component. There are docs on View components in the Layout section of the Forms docs. Inside that Blade view, you can render a button using the `<x-filament::button>` Blade component.

**Reply by danharrin** [reply]
If you put the buttons inside the form, they are not traditional actions like I originally suggested, those are just displayed at the end of the form.
